### PR TITLE
KOGITO-4991 - Trusty Kubernetes demonstration fix

### PR DESCRIPTION
**Pull request by r00ta**
JIRA: https://issues.redhat.com/browse/KOGITO-4991

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by r00ta** [comment]
Jenkins retest this please

File: trusty-demonstration/kubernetes/README.md [hunk 1]
```diff
@@ -47,41 +47,45 @@ minikube addons enable ingress
...
+```bash
+KOGITO_VERSION=v1.5.0
+```
+
 Deploy the kogito operator
 ```bash
-./kogito install operator
+wget https://github.com/kiegroup/kogito-operator/releases/download/${KOGITO_VERSION}/kogito-operator.yaml
+sed -i "s/kogito-operator-system/${PROJECT_NAME}/g" kogito-operator.yaml
+kubectl apply -f kogito-operator.yaml -n ${PROJECT_NAME}
```
Comment (danielezonca) [inline trusty-demonstration/kubernetes/README.md]: Is this necessary? If Kogito operator by default is installed in `namespace: kogito-operator-system` I don't know if it is safe to replace this

File: trusty-demonstration/kubernetes/README.md [hunk 2]
```diff
@@ -47,41 +47,45 @@ minikube addons enable ingress
...
 ```
 
 Install the [Infinispan operator](https://operatorhub.io/operator/infinispan/2.0.x/infinispan-operator.v2.0.6) version 2.0.6. 
 Pay attention that by default the Infinispan operator is installed on the namespace `my-infinispan`. You have to download the `yaml` file and change the namespace of the custom resources accordingly to your namespace.
 
+```bash
+wget https://operatorhub.io/install/2.0.x/infinispan.yaml
+sed -i "s/namespace: my-infinispan/namespace: ${PROJECT_NAME}/g" infinispan.yaml
+sed -i "s/- my-infinispan/- ${PROJECT_NAME}/g" infinispan.yaml
+kubectl apply -f infinispan.yaml -n ${PROJECT_NAME}
```
Comment (danielezonca) [inline trusty-demonstration/kubernetes/README.md:84]: Same comment here

File: trusty-demonstration/kubernetes/README.md [hunk 3]
```diff
@@ -138,6 +149,9 @@ spec:
   image: quay.io/<YOUR_NAMESPACE>/dmn-tracing-quarkus:2.0.0-snapshot # <---- replace with your image
   infra:
     - kogito-kafka-infra
+  env:
+    - name: KOGITO_SERVICE_URL
+      value: http://dmn-tracing-quarkus:8080
```
Comment (danielezonca) [inline trusty-demonstration/kubernetes/README.md:154]: Why this change? Is this a breaking change introduce with 1.5?

File: trusty-demonstration/kubernetes/README.md [hunk 4]
```diff
@@ -47,41 +47,45 @@ minikube addons enable ingress
...
 ```
 
-You should see that the `kogito-operator` has one pod running within some minutes: the command `kubectl get pods -n "operators"` should return something like 
+You should see that the `kogito-operator` has one pod running within some minutes: the command `kubectl get pods -n ${PROJECT_NAME}` should return something like 
 
 ```bash
-NAME                             READY   STATUS    RESTARTS   AGE
-kogito-operator-9f575f64-zp56t   1/1     Running   0          98s
+NAME                                                  READY   STATUS              RESTARTS   AGE
+kogito-operator-controller-manager-6bff9bb855-v7x9t   0/2     ContainerCreating   0          24s
```
Comment (jiripetrlik) [inline trusty-demonstration/kubernetes/README.md]: Maybe we should say that user should rather see both pods ready. Something like 2/2 instead of 0/2. Operator is working correctly only if pods are available.

File: trusty-demonstration/kubernetes/README.md [hunk 5]
```diff
@@ -94,14 +98,21 @@ find strimzi-${STRIMZI_VERSION}/install/cluster-operator -name '*RoleBinding*.ya
 kubectl apply -f strimzi-${STRIMZI_VERSION}/install/cluster-operator/ -n ${PROJECT_NAME}
 ```
 
-Deploy the `KogitoInfra` custom resources, so that the Kogito operator will take or managing the Infinispan and kafka resources.
+Deploy kafka and infinispan resources with 
+```bash
+kubectl apply -f resources/kafka.yaml -n ${PROJECT_NAME}
```
Comment (jiripetrlik) [inline trusty-demonstration/kubernetes/README.md:103]: Why this was not necessary before?

**Review by jiripetrlik — COMMENTED** [review]
I have few comments/questions.

File: trusty-demonstration/kubernetes/README.md (hunk 3 above)
Comment (r00ta) [reply trusty-demonstration/kubernetes/README.md:154]: Hi @danielezonca , this configuration was removed in another PR by mistake. It has been always needed

File: trusty-demonstration/kubernetes/README.md (hunk 5 above)
Comment (r00ta) [reply trusty-demonstration/kubernetes/README.md:103]: Hi @jiripetrlik , the operator is not provisioning anymore kafka and infinispan instances from v1.5. See 
Screenshot from 2021-04-21 10-18-31

File: trusty-demonstration/kubernetes/README.md (hunk 2 above)
Comment (r00ta) [reply trusty-demonstration/kubernetes/README.md:84]: hi @danielezonca , this is the way to install the infinispan operator in the selected namespace. Everything should be fine

File: trusty-demonstration/kubernetes/README.md (hunk 1 above)
Comment (r00ta) [reply trusty-demonstration/kubernetes/README.md]: Hi @danielezonca , this is the standard way to install the operator on a specific namespace 
Screenshot from 2021-04-26 15-25-07

Actually in the readme it is missing the fact that the namespace should be replaced in the yaml file (the command should fail). Could you confirm that this procedure is ok @ricardozanini ?

**Review by danielezonca — APPROVED** [review]
LGTM 👍

File: trusty-demonstration/kubernetes/README.md (hunk 1 above)
Comment (r00ta) [reply trusty-demonstration/kubernetes/README.md]: Sorry, I see only now that Ricardo is on PTO til next week. Do you mind to double check what I wrote above ^^ @radtriste ?

Comment (radtriste) [reply trusty-demonstration/kubernetes/README.md]: The `kogito-operator.yaml` file installs the Operator as cluster-wide operator and will listen to all namespaces in the cluster. See https://github.com/kiegroup/kogito-operator/blob/master/kogito-operator.yaml#L2983-L2986

If you want a namespace installation of the yaml, you would also need to change
```
- name: WATCH_NAMESPACE
  valueFrom:
    fieldRef:
      fieldPath: metadata.annotations['olm.targetNamespaces']
```
to
```
- name: WATCH_NAMESPACE
  value: <YOUR_NAMESPACE>
```

Comment (r00ta) [reply trusty-demonstration/kubernetes/README.md]: Thank you very much @radtriste for the double check. I've changed the instructions so to let the user install the operator cluster-wide.

**Review by radtriste — APPROVED** [review]

**Review by jiripetrlik — APPROVED** [review]
